stix2/v20/vocab.py

```diff
@@ -0,0 +1,178 @@
+"""
+STIX 2.0 open vocabularies and enums
+"""
+
+ATTACK_MOTIVATION = [
+    "accidental",
+    "coercion",
+    "dominance",
+    "ideology",
+    "notoriety",
+    "organizational-gain",
+    "personal-gain",
+    "personal-satisfaction",
+    "revenge",
+    "unpredictable",
+]
+
+
+ATTACK_RESOURCE_LEVEL = [
+    "individual",
+    "club",
+    "contest",
+    "team",
+    "organization",
+    "government",
+]
+
+
+HASHING_ALGORITHM = [
+    "MD5",
+    "MD6",
+    "RIPEMD-160",
+    "SHA-1",
+    "SHA-224",
+    "SHA-256",
+    "SHA-384",
+    "SHA-512",
+    "SHA3-224",
+    "SHA3-256",
+    "SHA3-384",
+    "SHA3-512",
+    "ssdeep",
+    "WHIRLPOOL",
+]
+
+
+IDENTITY_CLASS = [
+    "individual",
+    "group",
+    "organization",
+    "class",
+    "unknown",
+]
+
+
+INDICATOR_LABEL = [
+    "anomalous-activity",
+    "anonymization",
+    "benign",
+    "compromised",
+    "malicious-activity",
+    "attribution",
+]
+
+
+INDUSTRY_SECTOR = [
+    "agriculture",
+    "aerospace",
+    "automotive",
+    "communications",
+    "construction",
+    "defence",
+    "education",
+    "energy",
+    "entertainment",
+    "financial-services",
+    "government-national",
+    "government-regional",
+    "government-local",
+    "government-public-services",
+    "healthcare",
+    "hospitality-leisure",
+    "infrastructure",
+    "insurance",
+    "manufacturing",
+    "mining",
+    "non-profit",
+    "pharmaceuticals",
+    "retail",
+    "technology",
+    "telecommunications",
+    "transportation",
+    "utilities",
+]
+
+
+MALWARE_LABEL = [
+    "adware",
+    "backdoor",
+    "bot",
+    "ddos",
+    "dropper",
+    "exploit-kit",
+    "keylogger",
+    "ransomware",
+    "remote-access-trojan",
+    "resource-exploitation",
+    "rogue-security-software",
+    "rootkit",
+    "screen-capture",
+    "spyware",
+    "trojan",
+    "virus",
+    "worm",
+]
+
+
+REPORT_LABEL = [
+    "threat-report",
+    "attack-pattern",
+    "campaign",
+    "identity",
+    "indicator",
+    "intrusion-set",
+    "malware",
+    "observed-data",
+    "threat-actor",
+    "tool",
+    "vulnerability",
+]
+
+
+THREAT_ACTOR_LABEL = [
+    "activist",
+    "competitor",
+    "crime-syndicate",
+    "criminal",
+    "hacker",
+    "insider-accidental",
+    "insider-disgruntled",
+    "nation-state",
+    "sensationalist",
+    "spy",
+    "terrorist",
+]
+
+
+THREAT_ACTOR_ROLE = [
+    "agent",
+    "director",
+    "independent",
+    "infrastructure-architect",
+    "infrastructure-operator",
+    "malware-author",
+    "sponsor",
+]
+
+
+THREAT_ACTOR_SOPHISTICATION = [
+    "none",
+    "minimal",
+    "intermediate",
+    "advanced",
+    "expert",
+    "innovator",
+    "strategic",
+]
+
+
+TOOL_LABEL = [
+    "denial-of-service",
+    "exploitation",
+    "information-gathering",
+    "network-capture",
+    "credential-exploitation",
+    "remote-access",
+    "vulnerability-scanning",
+]
```

Comment:
From an API perspective I think it would be more useful for users to have individual variables for all labels rather than making them figure out what index the correct value has...

``` python
from stix2.v20 import Tool
from stix2.v20.vocabs import TOOL_LABEL

t = Tool(labels=[TOOL_LABELS[0]]
```

vs

``` python
from stix2.v20 import Tool
from stix2.v20.vocabs import TOOL_LABEL_DENIAL_OF_SERVICE

t = Tool(labels=[TOOL_LABEL_DENIAL_OF_SERVICE]
```

```suggestion
TOOL_LABEL_DENIAL_OF_SERVICE =  "denial-of-service"
TOOL_LABEL_EXPLOITATION = "exploitation"
TOOL_LABEL_INFORMATION_GATHERING = "information-gathering"
TOOL_LABEL_NETWORK_CAPTURE = "network-capture"
TOOL_LABEL_CREDENTIAL_EXPLOITATION = "credential-exploitation"
TOOL_LABEL_REMOTE_ACCESS = "remote-access"
TOOL_LABEL_VULNERABILITY_SCANNING = "vulnerability-scanning"

TOOL_LABEL = [
    TOOL_LABEL_DENIAL_OF_SERVICE,
    TOOL_LABEL_EXPLOITATION,
    TOOL_LABEL_INFORMATION_GATHERING,
    TOOL_LABEL_NETWORK_CAPTURE,
    TOOL_LABEL_CREDENTIAL_EXPLOITATION,
    TOOL_LABEL_REMOTE_ACCESS,
    TOOL_LABEL_VULNERABILITY_SCANNING,
]
```

Reply:
Well they wouldn't have to look up an index, they'd have to look up a variable name instead :)  I think the main benefit of this is more transparent code.  Just from reading the code, you couldn't tell which vocab value `TOOL_LABELS[0]` is.  But from `TOOL_LABEL_DENIAL_OF_SERVICE`, it's pretty obvious.  Another benefit is that if people use the variables instead of typing values themselves, they're less likely to make typos.

I'm willing to make variables if we want.

Reply:
I agree; I think we should make change.